### PR TITLE
fix: build pipeline

### PR DESCRIPTION
**Pull request by 0xAsimetriq**
# Breaking Changes

N/A

# Changes

- fix: Downgraded valtio to same version that we use in WalletConnect modal, see https://github.com/pmndrs/valtio/discussions/706 which is now an issue due to new scaffold-utils package
- fix: Turborepo builds
- fix: Invalid self import inside scafold-utils package
- chore: Separate `npm run watch` from `npm run laboratory` as running these in paralel caused out of order issues 
- chore: Update dependencies
- fix: Added missing dev-dependencies

# Associated Issues

closes #...

### CODE REVIEW

File: dangerfile.ts [hunk 1]
```diff
@@ -130,19 +130,17 @@ async function checkUiPackage() {
...
+  if (created_ui_composites_index_ts.length && !jsx_index_diff?.added.includes('../composites')) {
+    fail('New composites were added, but not exported in ui/utils/JSXTypeUtil.ts')
+  }
 
-  // if (created_ui_layout_index_ts.length && !jsx_index_diff?.added.includes('../layout')) {
-  //   fail('New layout components were added, but not exported in ui/utils/JSXTypeUtil.ts')
-  // }
+  if (created_ui_layout_index_ts.length && !jsx_index_diff?.added.includes('../layout')) {
+    fail('New layout components were added, but not exported in ui/utils/JSXTypeUtil.ts')
+  }
```
Comment (tomiir) [inline dangerfile.ts:143]: is this ok?

File: package.json [hunk 2]
```diff
@@ -4,23 +4,24 @@
   "private": true,
   "type": "module",
   "scripts": {
+    "build:clean": "turbo run build:clean",
     "build": "turbo run build",
-    "watch": "turbo run watch --parallel",
-    "gallery": "turbo run dev:gallery --parallel",
-    "laboratory": "turbo run dev:laboratory --parallel",
-    "examples": "turbo run dev:example --parallel",
+    "watch": "turbo run watch --concurrency=50 --continue",
```
Comment (tomiir) [inline package.json:9]: curious, why 50 in particular?

File: packages/scaffold-utils/tsconfig.json [hunk 3]
```diff
@@ -5,5 +5,5 @@
     "declarationDir": "./dist/types"
   },
   "extends": "../../tsconfig.json",
-  "include": ["index.ts", "src"]
+  "include": ["exports", "src"]
```
Comment (tomiir) [inline packages/scaffold-utils/tsconfig.json:8]: is this ok?

**Review by tomiir — COMMENTED** [review]
Looks good (though build broke), just some minor questions JIC!

File: packages/ethers5/exports/react.tsx [hunk 4]
```diff
@@ -28,10 +28,10 @@ export function createWeb3Modal(options: Web3ModalOptions) {
 
 // -- Hooks -------------------------------------------------------------------
 export function useWeb3ModalProvider() {
-  const state = useSnapshot(EthersStoreUtil.state)
+  const { provider, providerType } = useSnapshot(EthersStoreUtil.state)
 
-  const walletProvider = state.provider as ethers.providers.ExternalProvider | undefined
-  const walletProviderType = state.providerType
+  const walletProvider = provider as ethers.providers.ExternalProvider | undefined
+  const walletProviderType = providerType
```
Comment (Cali93) [inline packages/ethers5/exports/react.tsx:34]: Do we even need to reassign this ?

**Review by Cali93 — APPROVED** [review]
LGTM 👌

File: apps/laboratory/package.json [hunk 5]
```diff
@@ -11,17 +11,17 @@
...
+    "@web3modal/ethers": "3.5.0-6a05dc24",
+    "@web3modal/ethers5": "3.5.0-6a05dc24",
+    "@web3modal/wagmi": "3.5.0-6a05dc24",
+    "framer-motion": "10.16.16",
+    "next": "14.0.4",
     "next-auth": "4.24.5",
     "react-icons": "4.12.0",
     "siwe": "2.1.4",
-    "valtio": "1.12.1"
+    "valtio": "1.11.2"
```
Comment (enesozturk) [inline apps/laboratory/package.json:24]: Why downgrading? (Just to follow for future considerations) @0xAsimetriq

**Review by enesozturk — APPROVED** [review]
🔥

File: packages/ethers5/exports/react.tsx (hunk 4 above)
Comment (0xAsimetriq) [reply packages/ethers5/exports/react.tsx:34]: Not sure, I was only playing with the code here while debugging cc @svenvoskamp what do you think?

File: apps/laboratory/package.json (hunk 5 above)
Comment (0xAsimetriq) [reply apps/laboratory/package.json:24]: There is a valtio discussion linked in the description of pr. After removing cache, valtio version in `@walletconnect/modal` and ours started clashing and erroring. We'd need to update it there first then in wbe3modal.

**Review by svenvoskamp — DISMISSED** [review]

**Pull request merged**